client/src/main/java/org/perfrepo/client/PerfRepoClient.java

```diff
@@ -1,720 +1,107 @@
-//      if (permission == null || permission.getReportId() == null) {
-//         throw new IllegalArgumentException("Permission or its report ID cannot be null.");
-//      }
-//
-//      HttpPost post = createBasicPost("report/id/" + permission.getReportId() + "/deletePermission");
-//      setPostEntity(post, permission);
-//      HttpResponse resp = httpClient.execute(post);
-//      if (resp.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
-//         logHttpError("Error while deleting permission", post, resp);
-//         EntityUtils.consume(resp.getEntity());
-//         return false;
-//      }
-//      Header[] locations = resp.getHeaders(HttpHeaders.LOCATION);
-//      if (locations != null && locations.length > 0) {
-//         log.debug("Deleted permission at: " + locations[0].getValue());
-//      }
-//      EntityUtils.consume(resp.getEntity());
-//      return true;
-//   }
-//
-//   public String getServerVersion() throws Exception {
-//      HttpGet get = createBasicGet("info/version");
-//      HttpResponse resp = httpClient.execute(get);
-//      BufferedReader reader = new BufferedReader(new InputStreamReader(resp.getEntity().getContent()));
-//
-//      return reader.readLine();
-//   }
-//}
+/**
+ * PerfRepo
+ * <p>
+ * Copyright (C) 2015 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.perfrepo.client;
+
+import java.util.Collection;
+import org.apache.log4j.Logger;
+import org.perfrepo.dto.group.GroupDto;
+import org.perfrepo.dto.user.UserDto;
+import org.perfrepo.dto.util.authentication.AuthenticationResult;
+
+/**
+ * Performance Repository REST API Client.
+ *
+ * @author Michal Linhard (mlinhard@redhat.com)
+ * @author Jiri Holusa (jholusa@redhat.com)
+ */
+public class PerfRepoClient {
+
+    private static final Logger log = Logger.getLogger(PerfRepoClient.class);
+
+    private String username;
+    private String password;
+    private String repositoryUrl;
+    private AuthenticationResult token;
+
+    private TestClient testClient;
+    private TestExecutionClient testExecutionClient;
+    private ReportClient reportClient;
+    private AlertClient alertClient;
+
+    public PerfRepoClient(String username, String password, String repositoryUrl) {
+        this.username = username;
+        this.password = password;
+        this.repositoryUrl = repositoryUrl;
+    }
+
+    public void login() {
```

Comment:
I'm not sure I like the way it would work. You mean that before every request, I would have to call "login"? Doesn't sound very user friendly.